### PR TITLE
Move System.IO.Compression.DeflateStream lengthy remarks

### PR DESCRIPTION
**Pull request by carlossanlop**
Based on @gewarren suggestion [here](https://github.com/dotnet/runtime/pull/46716#issuecomment-761061215), I am moving all the lengthy remarks from DeflateStream.xml to their own md files.

### CODE REVIEW

File: includes/remarks/System.IO.Compression/DeflateStream/BeginRead.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Starting with the [!INCLUDE[net_v45](~/includes/net-v45-md.md)], you can perform asynchronous read operations by using the <xref:System.IO.Stream.ReadAsync%2A> method. The <xref:System.IO.Compression.DeflateStream.BeginRead%2A> method is still available in the [!INCLUDE[net_v45](~/includes/net-v45-md.md)] to support legacy code; however, you can implement asynchronous I/O operations more easily by using the new async methods. For more information, see [Asynchronous File I/O](/dotnet/standard/io/asynchronous-file-i-o).
```
Comment (gewarren) [inline includes/remarks/System.IO.Compression/DeflateStream/BeginRead.md:1]:
```suggestion
Starting with .NET Framework 4.5, you can perform asynchronous read operations by using the <xref:System.IO.Stream.ReadAsync%2A> method. The <xref:System.IO.Compression.DeflateStream.BeginRead%2A> method is still available in .NET Framework 4.5+ to support legacy code; however, you can implement asynchronous I/O operations more easily by using the new async methods. For more information, see [Asynchronous File I/O](/dotnet/standard/io/asynchronous-file-i-o).
```

File: includes/remarks/System.IO.Compression/DeflateStream/BeginWrite.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Starting with the [!INCLUDE[net_v45](~/includes/net-v45-md.md)], you can perform asynchronous write operations by using the <xref:System.IO.Stream.WriteAsync%2A> method. The <xref:System.IO.Compression.DeflateStream.BeginWrite%2A> method is still available in the [!INCLUDE[net_v45](~/includes/net-v45-md.md)] to support legacy code; however, you can implement asynchronous I/O operations more easily by using the new async methods. For more information, see [Asynchronous File I/O](/dotnet/standard/io/asynchronous-file-i-o).
```
Comment (gewarren) [inline includes/remarks/System.IO.Compression/DeflateStream/BeginWrite.md:1]:
```suggestion
Starting with .NET Framework 4.5, you can perform asynchronous write operations by using the <xref:System.IO.Stream.WriteAsync%2A> method. The <xref:System.IO.Compression.DeflateStream.BeginWrite%2A> method is still available in .NET Framework 4.5+ to support legacy code; however, you can implement asynchronous I/O operations more easily by using the new async methods. For more information, see [Asynchronous File I/O](/dotnet/standard/io/asynchronous-file-i-o).
```

File: includes/remarks/System.IO.Compression/DeflateStream/DeflateStream.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+This class represents the Deflate algorithm, which is an industry-standard algorithm for lossless file compression and decompression. Starting with the [!INCLUDE[net_v45](~/includes/net-v45-md.md)], the <xref:System.IO.Compression.DeflateStream> class uses the zlib library. As a result, it provides a better compression algorithm and, in most cases, a smaller compressed file than it provides in earlier versions of the .NET Framework.
```
Comment (gewarren) [inline includes/remarks/System.IO.Compression/DeflateStream/DeflateStream.md:1]:
```suggestion
This class represents the Deflate algorithm, which is an industry-standard algorithm for lossless file compression and decompression. Starting with .NET Framework 4.5, the <xref:System.IO.Compression.DeflateStream> class uses the zlib library. As a result, it provides a better compression algorithm and, in most cases, a smaller compressed file than it provides in earlier versions of .NET Framework.
```

**Review by gewarren — APPROVED** [review]

**Comment by carlossanlop** [comment]
Closing this in favor of https://github.com/dotnet/dotnet-api-docs/pull/5378, which contains _everything_ from System.IO.Compression.

**Pull request closed without merging**